src/kOS/Screen/GUIWindow.cs

```diff
@@ -0,0 +1,294 @@
+        public bool draggable = true;
+        private bool uiGloballyHidden = false;
+        private GUIWidgets widgets;
+        private GUIStyle style;
+        private GUIStyle commDelayStyle;
+        private Color noControlColor = new Color(1, 0.5f, 0.5f, 0.4f);
+        private Color slowControlColor = new Color(1, 0.95f, 0.95f, 0.9f);
+        private Texture2D commDelayedTexture;
+        public float extraDelay = 0f;
+
+        public bool ShowCursor { get; set; }
+
+        public void Awake()
+        {
+            // Transparent - leave the widget inside it to draw background if it wants to.
+            style = new GUIStyle(HighLogic.Skin.window);
+            style.normal.background = null;
+            style.onNormal.background = null;
+            style.focused.background = null;
+            style.margin = new RectOffset(0, 0, 0, 0);
+            style.padding = new RectOffset(0, 0, 0, 0);
+
+            commDelayStyle = new GUIStyle(HighLogic.Skin.label);
+            commDelayStyle.normal.textColor = Color.black;
+            commDelayStyle.alignment = TextAnchor.MiddleCenter;
+            commDelayStyle.clipping = TextClipping.Overflow;
+            commDelayStyle.stretchHeight = true;
+            commDelayStyle.fontSize = 10;
+            commDelayStyle.fontStyle = FontStyle.Bold;
+            commDelayedTexture = GameDatabase.Instance.GetTexture("kOS/GFX/commDelay", false);
+            var solidWhite = new Texture2D(1, 1);
+            solidWhite.SetPixel(0, 0, Color.white);
+            commDelayStyle.normal.background = solidWhite;
+
+            IsPowered = true;
+            WindowRect = new Rect(0, 0, 0, 0); // will get resized later in AttachTo().
+
+            GameEvents.onHideUI.Add (OnHideUI);
+			GameEvents.onShowUI.Add (OnShowUI);
+        }
+
+        public void OnDestroy()
+        {
+            if (shared != null) shared.RemoveWindow(this);
+            GameEvents.onHideUI.Remove(OnHideUI);
+            GameEvents.onShowUI.Remove(OnShowUI);
+        }
+        
+        void OnHideUI()
+        {
+            uiGloballyHidden = true;
+        }
+        
+        void OnShowUI()
+        {
+            uiGloballyHidden = false;            
+        }
+        
+        public override void GetFocus()
+        {
+        }
+        
+        public override void LoseFocus()
+        {
+        }
+
+        public override void Open()
+        {
+            base.Open();
+            BringToFront();
+        }
+
+        public void Toggle()
+        {
+            if (IsOpen) Close();
+            else Open();
+        }
+
+        void OnGUI()
+        {
+            if (!IsOpen) return;
+            
+            try
+            {
+                if (FlightResultsDialog.isDisplaying) return;
+                if (uiGloballyHidden && kOS.Safe.Utilities.SafeHouse.Config.ObeyHideUI) return;
+            }
+            catch(NullReferenceException)
+            {
+            }
```

Comment:
It looks like this was copied from TermWindow.cs, and was part of the initial commit by Nivekk.  That being said, it would probably be worthwhile to look into this behavior.  It may not be an issue at all anymore, or it may be possible to detect the null reference without throwing an exception (which is expensive, and particularly an issue inside of GUI update methods).  At the very least, catching the excepting on log a message of some kind (I'd prefer an error message) so that we can find out if this exception is ever being thrown.  I've checked and it looks like internally `FlightResultDialog.isDisplaying` includes a null check for the private `Instance` field.